### ISSUE TITLE
Personal preferences for `prevent-abbreviations`

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#1638]
I've gone back and forth on this rule and I don’t think its suggestions are always beneficial, even if ideal in practice. Here are my exclusions:

```json
 "replacements": {
   "db": false, // database is too long and db is super common
   "arg": false,
   "args": false, // arguments_ 🤢 
   "fn": false, // function_ 🤢 
   "func": {
     "fn": true,
     "function": false
   },

   // React
   "prop": false,
   "props": false,
   "ref": false,
   "refs": false
 },
 // These should be native exclusions
 "ignore": ["semVer", "SemVer"]
```

Feel free to close on sight, it's like, my opinion, man

**Comment by sindresorhus** [comment]
I have been using this rule for a long time and I honestly feel it's ideal in practice too. We knew when we made the rule that many would object. JS has historically been pretty bad at encouraging readable code, and there are a lot of conventions that are hard to turn.

**Comment by sindresorhus** [comment]
> // These should be native exclusions

Agreed. You can add them here: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/32c20196528c28d9dd6eb1f3beb15f5528480418/rules/shared/abbreviations.js#L243-L248

**Comment by sindresorhus** [comment]
> // React

I guess we could document this for React users.

**Comment by sindresorhus** [comment]
> // database is too long and db is super common

The writing part of coding is the easy part. It's not gonna make you more efficient to use `db` instead of `database`.

**Comment by futpib** [comment]
We could also detect react or provide a preset for react.

**Comment by fregante** [comment]
> The writing part of coding is the easy part. It's not gonna make you more efficient to use `db` instead of `database`.

Nobody reads `db` in a file and doesn't know what it refers to.

**Comment by sindresorhus** [comment]
The great part about this rule is that it's totally configurable and it's easy to hide the overrides in a shareable config.

**Comment by fregante** [comment]
> The great part about this rule is that it's totally configurable and it's easy to hide the overrides in a shareable config.

🤷‍♂️

**Comment by sindresorhus** [comment]
I read it as "douche bag" and get offended every time :P (not)

**Comment by fregante** [comment]
I see the `_` in `function_` as excited body part and I'd rather see a cute `fn` instead.

**Comment by sindresorhus** [comment]
> We could also detect react or provide a preset for react.

I wonder if we could just detect React being used in a file and disable those in the rule itself. It would be nicer if the rule would just handle it rather than needing a preset.

**Comment by fisker** [comment]
I have a suggestion, we can ignore package names and  other shape of the package name, eg camel case. This will also free you from using `semver`.

**Comment by sindresorhus** [comment]
> This will also free you from using semver.

We should still ignore `semver` though as it can be used without using the actual `semver` package.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1639

**Comment by sindresorhus** [comment]
SemVer is just a name. It can be used in other contexts to refer to semantic versioning.

```js
const semver = '1.2.4';

t.is(getPackageVersion(), semver);
```

**Comment by fregante** [comment]
Closing this. The React part is tracked in

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1639
